include/lang_utils/tuple.h

```diff
@@ -25,52 +26,41 @@ class transform_tuple_type {
                           std::tuple_size<ORIG>::value>>::type;
 };
 
-template <typename ...TUPLES>
-struct tuple_sizes_equal;
+template <size_t N>
+constexpr size_t check_array_equal(size_t const (&arr)[N]) {
+    size_t ret = 0;
+    for (size_t s : arr) {
+        ret = (s == arr[0]) ? s : throw "Sizes mismatch";
+    }
+    return ret;
+}
 
-template <typename TUPLE>
-struct tuple_sizes_equal<TUPLE>{
-    static const size_t value = std::tuple_size<typename std::decay<TUPLE>::type>::value;
-};
+template <typename... TUPLES>
+static constexpr size_t const tuple_sizes_equal_v =
+    check_array_equal({std::tuple_size_v<std::decay_t<TUPLES>>...});
```

Comment:
I have found that `static_assert` does not actually prevent any of the compile errors that would result from doing what it tries to prevent. And also yes, less code.